src/OrbitGl/App.cpp

```diff
@@ -1694,34 +1694,54 @@ orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadSymbols(
   return result_future;
 }
 
-ErrorMessageOr<const ModuleData*> OrbitApp::GetLoadedModuleByPath(const std::string& module_path) {
-  std::optional<ModuleInMemory> module_in_memory =
-      GetTargetProcess()->FindModuleByPath(module_path);
-  if (!module_in_memory.has_value()) {
-    return ErrorMessage{absl::StrFormat("Module \"%s\" is not loaded by process.", module_path)};
-  }
+ErrorMessageOr<std::vector<const ModuleData*>> OrbitApp::GetLoadedModulesByPath(
+    const std::filesystem::path& module_path) {
+  std::vector<std::string> build_ids =
+      GetTargetProcess()->FindModuleBuildIdsByPath(module_path.string());
+
+  std::vector<const ModuleData*> result;
+  for (const auto& build_id : build_ids) {
+    const ModuleData* module_data =
+        module_manager_->GetModuleByPathAndBuildId(module_path.string(), build_id);
+    if (module_data == nullptr) {
+      ERROR("module \"%s\" was loaded by the process, but is not part of module manager",
+            module_path.string());
+      crash_handler_->DumpWithoutCrash();
+      return ErrorMessage{"Unexpected error while loading preset."};
+    }
 
-  const ModuleData* module_data =
-      module_manager_->GetModuleByPathAndBuildId(module_path, module_in_memory->build_id());
-  if (module_data == nullptr) {
-    ERROR("module \"%s\" was loaded by the process, but is not part of module manager",
-          module_path);
-    crash_handler_->DumpWithoutCrash();
-    return ErrorMessage{"Unexpected error while loading preset."};
+    result.emplace_back(module_data);
   }
 
-  return module_data;
+  return result;
 }
 
-orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadPresetModule(const std::string& module_path,
-                                                                    const PresetFile& preset_file) {
-  auto module_data_or_error = GetLoadedModuleByPath(module_path);
+orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadPresetModule(
+    const std::filesystem::path& module_path, const PresetFile& preset_file) {
+  auto modules_data_or_error = GetLoadedModulesByPath(module_path);
 
-  if (module_data_or_error.has_error()) {
-    return module_data_or_error.error();
+  if (modules_data_or_error.has_error()) {
+    return modules_data_or_error.error();
   }
 
-  const ModuleData* module_data = module_data_or_error.value();
+  std::vector<const ModuleData*>& modules_data = modules_data_or_error.value();
+
+  if (modules_data.empty()) {
+    return ErrorMessage{
+        absl::StrFormat("Module \"%s\" is not loaded by process.", module_path.string())};
+  }
+
+  // TODO(b/191240539): Ask the user which build_id they prefer.
+  if (modules_data.size() > 1) {
```

Comment:
Just for my understanding: What would be the use-case for this? I assume we expect not to see this situation very often, only with esoteric module setups, right?

Reply:
if a module was reloaded on the fly but mapping for old version of the module is still there (and it might be used for some time). This is not normal use-case as far as I can speculate.

Reply:
Thanks. Makes sense.